src/fmu/tools/utilities/sample_attributes.py

```diff
@@ -0,0 +1,162 @@
+"""Sample attributes on grid resolution as dataframe points sets.
+
+This usage is for setting attributes on grid resolution, e.g. a seismic attribute
+(from a map) combined with a region parameter from the same 3D grid.
+
+This is targeted to the "sim2seis" workflow in FMU.
+"""
+
+from __future__ import annotations
+
+import logging
+import pathlib
+import tempfile
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Literal
+
+import numpy as np
+import xtgeo
+
+if TYPE_CHECKING:
+    import pandas as pd
+
+# custom logger for this module
+logger = logging.getLogger(__name__)
+
+
+class Attrs(str, Enum):
+    OBS = "OBS"
+    OBS_ERROR = "OBS_ERROR"
+    REGION = "REGION"
+    VALUES = "VALUES"
+
+
+def _get_layer(position: tuple, grid: xtgeo.Grid, zone: xtgeo.GridProperty) -> int:
+    """Get the layer index (base=1!) based on the position and zone."""
+    zone_name, where = position
+
+    if zone and zone_name:
+        if zone_name not in zone.codes.values():
+            raise ValueError(f"Zone name '{zone_name}' not found in the grid.")
+
+        nzone = {v: k for k, v in zone.codes.items()}.get(zone_name)
+        logger.debug("Zone index to apply for %s is: %s", zone_name, nzone)
+
+        indices = np.where(zone.values == nzone)
+        upper, lower = indices[2].min() + 1, indices[2].max() + 1  # base=1
+
+        return {"top": upper, "base": lower}.get(where, round(0.5 * (upper + lower)))
```

Comment:
We should check if the input `where` is within the three options `top/base/center` and raise error if not. And it would be better to be more explicit in this return to understand that the option `center` returns `round(0.5 * (upper + lower)`